docs/dev/migrating-to-v1.rst

```diff
@@ -183,3 +183,68 @@ has not been changed compared to the GridModel.
 | model.setup_grid(...)        | model.grid_component.create(...)          |
 | model.setup_grid_from_*(...) | model.grid_component.add_data_from_*(...) |
 +------------------------------+-------------------------------------------+
+
+
+DataAdapter
+-----------
+
+The previous version of the `DataAdapter` and it's subclasses had a lot of
+responsabilities:
+- Validate the input from the `DataCatalog` entry.
+- Find the right paths to the data based on a naming convention.
+- Deserialize/read many different file formats into python objects.
+- Merge these different python objects into one that represent that data source in the
+model region.
+- Homogenize the data based on the data catalog entry and HydroMT conventions.
+
+In v1, this class has been split into three extentable components:
+
+DataSource
+^^^^^^^^^^
+
+The `DataSource` is the python representation of a parsed entry in the `DataCatalog`.
+The `DataSource` is responsable for validating the `DataCatalog` entry. It also carries
+the soon-to-be introduced `DataAdapter` and `DataDriver` and serves as an entrypoint to
+the data.
+Per HydroMT data type (e.g. `RasterDataset`, `GeoDataFrame`), HydroMT has one
+`DataSource`, e.g. `RasterDatasetSource`, `GeoDataFrameSource`.
+
+MetaDataResolver
```

Comment:
Idea in the beginning was to read metadata coming from querying the uri to learn more. But indeed, we are resolving based on the uri, so that could be a good name as well.